tests/unit_test/fuel/utils/json_config_test.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+
+from nvflare.fuel.utils.config import ConfigFormat
+from nvflare.fuel.utils.json_config_loader import JsonConfigLoader
+
+
+class TestJsonConfig:
+
+    def return_dict(self, file_name):
+        if file_name == "test.json":
+            return {
+                "a": {
+                    "a1": 1,
+                    "a2": 2,
+                },
+                "b": 1,
+                "c": "hi",
+                "d": [1, 2]
+            }
+        else: # default
+            return {
+                "a": {
+                    "a1": 2,
+                    "a2": 4,
+                },
+                "b": 2,
+                "c": "hello",
+                "d": [2, 4]
+            }
+
+    def test_json_loader(self):
+        loader = JsonConfigLoader()
+        loader._from_file = self.return_dict
+        dicts = {
+            "a": {
+                "a1": 200,
+            },
+            "c": "hello",
+            "d": [200, 400, 500],
+            "e1": "Yes",
+            "e2": "True",
+            "e3": "NO",
+        }
+
+        config = loader.load_config("test.json", "default_test.json", dicts)
+        assert config is not None
+        conf = config.get_native_conf()
+        assert conf["a"]["a1"] == 200
+        assert conf.get("b") == 1
+        assert conf.get("c") == "hello"
+        assert conf.get("d") == [200, 400, 500]
+        assert conf.get("e1") == "Yes"
+        assert conf.get("e2") == "True"
+        assert conf.get("e3") == "NO"
+        assert conf.get("e4", None) is None
```

Comment:
Would "assert conf == dicts" achieve the same effect?

Reply:
I re-read again and found your loader.load_config not only loads config from file it also loads from dict and it also merges them.
Given that implementation this test case is good.